util/http.py

```diff
@@ -257,3 +260,56 @@ def _process_response(cls, url, response, allowed_response_codes=None,
             )
         return response
 
+    @classmethod
+    def debuggable_get(cls, url, **kwargs):
+        """Make a GET request that returns a detailed problem
+        detail document on error.
+        """
+        return cls.debuggable_request("GET", url, **kwargs)
+
+    @classmethod
+    def debuggable_post(cls, url, payload, **kwargs):
+        """Make a POST request that returns a detailed problem
+        detail document on error.
+        """
+        kwargs['data'] = payload
+        return cls.debuggable_request("POST", url, **kwargs)
+
+    @classmethod
+    def debuggable_request(cls, http_method, url, **kwargs):
+        """Make a request that returns a detailed problem detail document on
+        error, rather than a generic "an integration error occured"
+        message.
+        """
+        if not 'allowed_response_codes' in kwargs:
+            # We allow all response codes so that we can handle bad
+            # ones in a more helpful way.
+            kwargs['allowed_response_codes']=["1xx", "2xx", "3xx", "4xx", "5xx"]
+        response = HTTP.request_with_timeout(http_method, url, **kwargs)
+        return cls.process_debuggable_response(response)
+
+    @classmethod
+    def process_debuggable_response(cls, response):
+        """If there was a problem with an integration request,
+        return an appropriate ProblemDetail. Otherwise, return the
+        response to the original request.
+
+        :param response: A Response object from the requests library.
+        """
+        content_type = response.headers.get('Content-Type')
+        if content_type == PROBLEM_DETAIL_JSON_MEDIA_TYPE:
+            return INTEGRATION_ERROR.detailed(
+                _('Remote service returned a problem detail document: %r') % (
+                    response.content
+                )
+            )
+        elif response.status_code / 100 not in (2,3):
```

Comment:
It might be useful to keep the argument for allowed response codes - I could imagine a case where you'd expect a 401 but want to use this in case you got something else.